pdata/pprofile/attributes_test.go

```diff
@@ -44,34 +44,46 @@ func TestFromAttributeIndices(t *testing.T) {
 	assert.Equal(t, attrs.AsRaw(), m)
 }
 
-func TestAddAttribute(t *testing.T) {
+func TestPutAttribute(t *testing.T) {
 	table := NewAttributeTableSlice()
-	att := table.AppendEmpty()
-	att.SetKey("hello")
-	att.Value().SetStr("world")
-
-	// Add a brand new attribute
-	loc := NewLocation()
-	err := AddAttribute(table, loc, "bonjour", pcommon.NewValueStr("monde"))
-	require.NoError(t, err)
-
+	indices := NewProfile()
+
+	// Put a first attribute.
+	require.NoError(t, PutAttribute(table, indices, "hello", pcommon.NewValueStr("world")))
```

Comment:
The previous test was checking against several data types as an indirect way to test that all the proper types were supported.
I agree this wasn't super clear. But I don't think we should lose that.

Reply:
Can you clarify that you are referring to `Location` and `Mapping` as `attributable`?
If so, I'd suggest to test *all* the pprofile data types that have attributes.

Reply:
Yes, that's what I refer to. Maybe use subtests?

Reply:
See new commit.